src/compiler/utilities.ts

```diff
@@ -2732,6 +2732,10 @@ namespace ts {
         return "__@" + symbolName as __String;
     }
 
+    export function getPropertyNameForPrivateNameDescription(containingClassSymbol: Symbol, description: __String): __String {
+        return `__#${getSymbolId(containingClassSymbol)}@${description}` as __String;

```

Comment:
For consistency we should probably pull the name formatting logic for unique symbols out of checker.ts (in `getLateBoundNameFromType`) and put it in a neighboring function here as well. Something like:

```ts
export function getPropertyNameForUniqueESSymbol(symbol: Symbol): __String {
    return `__@${getSymbolId(symbol)}@${symbol.escapedName}` as __String;
}
```

(note that this reverses the order from the one in checker to be consistent with the above declaration).